src/components/pages/address/address-events.tsx

```diff
@@ -1,13 +1,108 @@
-import { Address } from "viem";
-import { Card, CardHeader, CardContent } from "@/components/ui/card";
-
-const AddressEvents = ({ address }: { address: Address }) => {
-  return (
-    <Card>
-      <CardHeader>Latest 25 Contract Events</CardHeader>
-      <CardContent>CONTRACT EVENTS</CardContent>
-    </Card>
-  );
+
+import Link from "next/link";
+import { Address, Block } from "viem";
+import { abiFromBytecode, whatsabi } from "@shazow/whatsabi";
+
+import { l2PublicClient } from "@/lib/chains";
+import { formatEventLog, formatTimestamp } from "@/lib/utils";
+
+import { Table, TableHeader, TableRow, TableHead, TableBody, TableCell } from "@/components/ui/table";
+import { DecodedLogDisplay } from "@/components/ui/decodedLog-display"
```

Comment:
`components/ui` is only for shadcn/ui components, move your DecodedLogDisplay component under pages/address.